cmake/aws-checksums/CMakeLists.txt

```diff
@@ -0,0 +1,66 @@
+cmake_minimum_required (VERSION 3.1)
+
+project (aws-checksums C)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/aws-checksums/cmake")
+include(AwsSharedLibSetup)
+
+file(GLOB AWS_CHECKSUMS_HEADERS
+     "aws-checksums/include/aws/checksums/*.h"
+)
+list(FILTER AWS_CHECKSUMS_HEADERS EXCLUDE REGEX ".*jni.*")
+
+file(GLOB AWS_CHECKSUMS_PRIV_HEADERS
+     "aws-checksums/include/aws/checksums/private/*.h"
+)
+
+file(GLOB AWS_CHECKSUMS_SRC
+    "aws-checksums/source/*.c"
+)
+list(FILTER AWS_CHECKSUMS_SRC EXCLUDE REGEX ".*jni.*")
+
+file(GLOB AWS_ARCH_SRC
+    "aws-checksums/source/intel/*.c"
+    "aws-checksums/source/arm/*.c"
+)
+
+file(GLOB CHECKSUMS_HEADERS
+    ${AWS_CHECKSUMS_HEADERS}
+    ${AWS_CHECKSUMS_PRIV_HEADERS}
+)
+
+file(GLOB CHECKSUMS_SRC
+  ${AWS_CHECKSUMS_SRC}
+    ${AWS_CHECKSUMS_PLATFORM_SOURCE}
+    ${AWS_ARCH_SRC}
+)
+
+add_library(aws-checksums SHARED ${CHECKSUMS_HEADERS} ${CHECKSUMS_SRC})
+target_compile_options(aws-checksums PRIVATE ${_FLAGS})
+set_target_properties(aws-checksums PROPERTIES LINKER_LANGUAGE C)
+set(CMAKE_C_FLAGS_DEBUGOPT "")
+set_property(TARGET aws-checksums PROPERTY C_STANDARD 99)
+
+target_compile_options(aws-checksums PRIVATE -fPIC -O2)
+
+target_include_directories(${CMAKE_PROJECT_NAME} PUBLIC
```

Comment:
this is not correct. ideally, should use the name of the inner most project, namely `PROJECT_NAME`.

Reply:
> this is not correct. ideally, should use the name of the inner most project, namely `PROJECT_NAME`.

i had to do this [ugly hack](https://github.com/ceph/aws-sdk-cpp/pull/2/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR240) to resolve this.
will try your suggestion :-)